kaspresso/src/main/kotlin/com/kaspersky/kaspresso/kaspresso/Kaspresso.kt

```diff
@@ -93,36 +53,35 @@ import com.kaspersky.kaspresso.interceptors.watcher.view.impl.logging.LoggingVie
 import com.kaspersky.kaspresso.interceptors.watcher.view.impl.logging.LoggingWebAssertionWatcherInterceptor
 import com.kaspersky.kaspresso.logger.UiTestLogger
 import com.kaspersky.kaspresso.logger.UiTestLoggerImpl
-import com.kaspersky.kaspresso.params.AutoScrollParams
-import com.kaspersky.kaspresso.params.ContinuouslyParams
-import com.kaspersky.kaspresso.params.FlakySafetyParams
-import com.kaspersky.kaspresso.params.Params
-import com.kaspersky.kaspresso.params.StepParams
+import com.kaspersky.kaspresso.params.*
 import com.kaspersky.kaspresso.report.impl.AllureReportWriter
 import com.kaspersky.kaspresso.testcases.core.testcontext.BaseTestContext
+import io.github.kakaocup.kakao.Kakao
 
 /**
  * The storage of all Kaspresso preferences and entities, such as [AdbServer], [Device] and different interceptors.
  */
 data class Kaspresso(
     internal val libLogger: UiTestLogger,
     internal val testLogger: UiTestLogger,
-    internal val adbServer: AdbServer,
-    internal val device: Device,
     internal val params: Params,
     internal val viewActionWatcherInterceptors: List<ViewActionWatcherInterceptor>,
     internal val viewAssertionWatcherInterceptors: List<ViewAssertionWatcherInterceptor>,
     internal val atomWatcherInterceptors: List<AtomWatcherInterceptor>,
     internal val webAssertionWatcherInterceptors: List<WebAssertionWatcherInterceptor>,
-    internal val objectWatcherInterceptors: List<ObjectWatcherInterceptor>,
-    internal val deviceWatcherInterceptors: List<DeviceWatcherInterceptor>,
     internal val viewBehaviorInterceptors: List<ViewBehaviorInterceptor>,
     internal val dataBehaviorInterceptors: List<DataBehaviorInterceptor>,
     internal val webBehaviorInterceptors: List<WebBehaviorInterceptor>,
+    internal val stepWatcherInterceptors: List<StepWatcherInterceptor>,
+    internal val testRunWatcherInterceptors: List<TestRunWatcherInterceptor>,
+
+    //Kautomator dependent
+    internal val adbServer: AdbServer,
+    internal val device: Device?,
```

Comment:
Remark:
Tests that are marked as shared tests, but contain some Kautomator code (e.g. use a UiScreen), will fail with NullPointerException since Device is null for them.
Would be good to catch that exception and return a more meaningful error.

Any ideas on what is the best way to catch that error and what to show as error message are welcome :)